File: JJSwiftClass_02/03-block回调的回顾/03-block回调的回顾/HttpTool.m
//
//  HttpTool.m
//  03-block回调的回顾
//
//  Created by xiaomage on 15/11/29.
//  Copyright © 2015年 xiaomage. All rights reserved.
//

#import "HttpTool.h"

@interface HttpTool ()

/** 回调block */
@property (nonatomic, copy) void(^callBack)();

@end

@implementation HttpTool

- (void)requestData:(void (^)())callBack
{
    self.callBack = callBack;
    
    dispatch_async(dispatch_get_global_queue(0, 0), ^{
        
        NSLog(@"正在进行网络请求:%@",[NSThread currentThread]);
        
        dispatch_async(dispatch_get_main_queue(), ^{
            callBack();
        });
    });
}

@end

File: JJSwiftClass_02/04-闭包的简单使用/04-闭包的简单使用/HttpTool.swift
//
//  HttpTool.swift
//  04-闭包的简单使用
//
//  Created by xiaomage on 15/11/29.
//  Copyright © 2015年 xiaomage. All rights reserved.
//

import UIKit

class HttpTool: NSObject {
    
    var callBack : (()->())?
    
    // 闭包的类型:(参数列表)->(返回值)
    // 建议:以后写闭包类型直接:()->()
    func requestData(_ callBack : @escaping ()->()) {
        
        self.callBack = callBack
        
        DispatchQueue.global(priority: 0).async { () -> Void in
            print("正在网络请求:\(Thread.current)")
            
            DispatchQueue.main.async(execute: { () -> Void in
                callBack()
            })
        }
    }
}
